### ISSUE TITLE
Change Path of stored GME to NAS(SMB/NFS/AFP share)

### ISSUE THREAD

**Issue by dirtyheizer** [thawn/ttmp32gme#17]
Gents,
I was curious wondering on how to change the path where the GME files are stored.
Do you know by chance how to edit oder re-link the database to a NAS drive?
Out of the box uses /Users/username/library
Thanks.

**Comment by thawn** [comment]
@dirtyheizer This is currently not possible unless you change the library path in the source code (for Mac OSX this is src/TTMp32Gme/Build/Mac.pm line 21
However this would require you to either runt the app from source or compile it on your own machine.

**Comment by thawn** [comment]
I have labeled this as a feature request. However, I have no idea if or when I will have the time to implement this.

**Comment by thawn** [comment]
Pull requests implementing a configurable library path are of course always appreciated :)

**Comment by thawn** [comment]
As a workaround, you can manually move the gme files to your NAS. This only means that you cannot use the "copy to tiptoi" function from within ttmp32gme but have to manually copy the gme files to the tiptoi from your NAS.

**Comment by thawn** [comment]
If you use [the docker image](https://hub.docker.com/r/thawn/ttmp32gme) you can choose where the library will be located:
```
ttmp32gmeStorage=</path/where/you/want/the/data>
tiptoi=</path/where/your/tiptoi/is/nounted>

docker run -d \
		--rm \
    	--publish 8080:8080 \
	--volume ${ttmp32gmeStorage}:/var/lib/ttmp32gme \
        --volume ${tiptoi}:/mnt/tiptoi
    	--name ttmp32gme \
    	thawn/ttmp32gme:latest
```

**Comment by acetous** [comment]
I'm on Windows and would like to store the library to a different drive, too. Futher more I think this is a common use case for all "power users" of this tool. ;)

Maybe this does not need to be an option in the "configuration" dialog. Changing this option could be some sort of work, like copying the existing library to the new path. The tool could read a confguration file inside the working directory. Therefore "power users" could just change some settings there without the need support this feature in your frontend.

**Comment by thawn** [comment]
@acetous Thank you very much for your suggestion. As you can see, making this path configurable is already on my todo list. Unfortunately, since I am working on this project in my free time, I cannot promise when I will have time to implement this feature. But the project is open source - I always welcome pull requests (which is how we got Docker support).

For power users: on linux/OSX, I would solve this problem with a symbolic link.I have never tried this, but I believe [symbolic links also work under windows](https://www.howtogeek.com/howto/16226/complete-guide-to-symbolic-links-symlinks-on-windows-or-linux/).

As to your suggestion with a less obvious configuration option: Currently the configuration is stored in a sqlite database. Adding support for another config file would be as much work as implementing a proper configuration option in the UI.

**Comment by thawn** [comment]
The latest source version now contains a configuration option to store the library in a different location. Once you change the location. The respective folder is created and the library is moved there.
This feature will be available in release version 0.3.0
